e2e/collector_test.go

```diff
@@ -24,405 +24,495 @@ import (
 
 	gcm "cloud.google.com/go/monitoring/apiv3/v2"
 	gcmpb "cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
-	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kubeutil"
-	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/operatorutil"
+	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
 	"github.com/google/go-cmp/cmp"
 	"google.golang.org/api/iterator"
 	"google.golang.org/protobuf/types/known/timestamppb"
-	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-
-	// Blank import required to register GCP auth handlers to talk to GKE clusters.
-	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/kubernetes"
 
 	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
-	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/generated/clientset/versioned"
 )
 
-func TestCollector(t *testing.T) {
-	t.Parallel()
-	tctx := newOperatorContext(t)
+func TestCollectorPodMonitoring(t *testing.T) {
+	ctx := context.Background()
+	kubeClient, opClient, err := newKubeClients()
+	if err != nil {
+		t.Fatalf("error instantiating clients. err: %s", err)
+	}
 
 	// We could simply verify that the full collection chain works once. But validating
 	// more fine-grained stages makes debugging a lot easier.
-	t.Run("deployed", tctx.subtest(testCollectorDeployed))
-
-	t.Run("scrape", tctx.subtest(func(ctx context.Context, t *OperatorContext) {
-		t.createOperatorConfigFrom(ctx, monitoringv1.OperatorConfig{
-			Features: monitoringv1.OperatorFeatures{
-				TargetStatus: monitoringv1.TargetStatusSpec{
-					Enabled: true,
+	t.Run("collector-deployed", testCollectorDeployed(ctx, t, kubeClient))
+	t.Run("collector-configured", testCollectorConfigured(ctx, t, kubeClient, opClient))
```

Comment:
Same problem as in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/747#discussion_r1474426870, maybe merge with `-configuration` job?

Reply:
Thanks! Renamed to emphasize we're updating the operatorconfig.